packages/web/src/graphql/withCell.tsx

```diff
@@ -43,38 +62,47 @@ export const withCell = ({
   }),
   QUERY,
   afterQuery = (data) => ({ ...data }),
-  Loading = () => 'Loading...',
+  Loading = () => <div>Loading...</div>,
   Failure,
   Empty,
   Success,
+}: {
+  beforeQuery?: (props: OperationVariables) => BaseQueryOptions
+  QUERY: DocumentNode | ((before: BaseQueryOptions) => DocumentNode)
+  afterQuery?: (data: DataObject) => DataObject
+  Loading?: React.FC<CellLoadingEmptyStateComponent>
+  Failure?: React.FC<CellFailureStateComponent>
+  Empty?: React.FC<CellLoadingEmptyStateComponent>
+  Success: React.FC<CellSuccessStateComponent>
 }) => {
-  const isDataNull = (data) => {
+  const isDataNull = (data: DataObject) => {
     return dataField(data) === null
   }
 
-  const isDataEmptyArray = (data) => {
-    return Array.isArray(dataField(data)) && dataField(data).length === 0
+  const isDataEmptyArray = (data: DataObject) => {
+    const field = dataField(data)
+    return Array.isArray(field) && field.length === 0
   }
 
-  const dataField = (data) => {
+  const dataField = (data: DataObject) => {
     return data[Object.keys(data)[0]]
   }
 
-  const isEmpty = (data) => {
+  const isEmpty = (data: DataObject) => {
     return isDataNull(data) || isDataEmptyArray(data)
   }
 
-  return (props) => (
+  return (props: OperationVariables) => (
     <Query
       query={typeof QUERY === 'function' ? QUERY(beforeQuery(props)) : QUERY}
       {...beforeQuery(props)}
     >
-      {({ error, loading, data, ...queryRest }) => {
+      {({ error, loading, data, ...queryRest }: QueryResultAlias) => {
         if (error) {
           if (Failure) {
             return <Failure error={error} {...queryRest} {...props} />
           } else {
-            throw new Error(error)
+            throw new Error((error as unknown) as string)
```

Comment:
`error` is of type `ApolloError`, so it wouldn't let me pass it as is. Another way to do it would be

```suggestion
            throw new Error(error.message)
```

what do you think would be better?

Reply:
Could you re-throw ⛹️‍♀️ the original error: `throw error`